Community/ideAlarm/automation/lib/python/community/idealarm/__init__.py

```diff
@@ -304,16 +305,21 @@ def onToggleSwitch(self, itemName):
         Called whenever an alarm arming mode toggle switch has been switched.
         '''
         newArmingMode = None
+        pinCode = getItemValue(itemName, 0)
         if itemName == self.armAwayToggleSwitch:
             if self.getArmingMode() in [ARMINGMODE['DISARMED']]:
                 newArmingMode = ARMINGMODE['ARMED_AWAY']
-            else:
+            elif pinCode == self.pinCode or self.pinCode == None:
```

Comment:
While this might work, the proper way to compare to None is by using 'is'...

    elif pinCode == self.pinCode or self.pinCode is None:

Reply:
Fixed